### DISCUSSION TITLE
Notable plugin succeeds rendering "An error occurred"

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#895] [General]
https://github.com/jayvdb/jayvdb/blob/09a74c9/notable.svg is "An error occurred", but the job https://github.com/jayvdb/jayvdb/runs/5225974781?check_suite_focus=true shows no details of the error.

**Comment by lowlighter** [comment]
I'm tagging your job step (https://github.com/jayvdb/jayvdb/blob/c5e27ef16d47336803f8909b48f571ad7d816e4d/.github/workflows/main.yml#L43-L67) to investigate on it later 👍

**Reply by jayvdb** [reply]
This happened again recently https://github.com/jayvdb/jayvdb/commit/669222b998888d7c9d5edf9cc9c0cce676db2a68 (has since fixed itself), and I've seen it has occurred a few other times since I first raised this.

I'd would be handy if the job didnt succeed, and didnt write the svg, as I would prefer a slightly out of date render than "An error occurred" to be appearing on the output for days at a time.

Also it would be helpful if the commit message can include the actions run id, so it is easy to determine which log needs to be looked at for a given commit.

**Reply by lowlighter** [reply]
Seems to be a query timeout:
https://github.com/jayvdb/jayvdb/runs/7367761366?check_suite_focus=true#step:3:835

Probably need to catch it better and retry/ignore

> I'd would be handy if the job didnt succeed, and didnt write the svg, as I would prefer a slightly out of date render than "An error occurred" to be appearing on the output for days at a time.

You can set [`plugins_errors_fatal: yes`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#plugins_errors_fatal) to make the workflow error instead of displaying error in the SVG

When this setting is on [`retries`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#retries) should also take effect and try thrice (by default) before giving up

> Also it would be helpful if the commit message can include the actions run id, so it is easy to determine which log needs to be looked at for a given commit.

Yes this is a good idea. I think the `pull-request` method already so it makes sense to include it in the commit too

**Reply by jayvdb** [reply]
Thanks. I added `plugins_errors_fatal: true` which worked well - I got an error. And then I set `repositories_batch = 5` to get it green again.
